core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessageDuplicator.java

```diff
@@ -44,129 +44,79 @@
 import com.google.common.base.MoreObjects.ToStringHelper;
 import com.spotify.futures.CompletableFutures;
 
-import com.linecorp.armeria.common.CommonPools;
-import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.ContentTooLargeException;
 import com.linecorp.armeria.common.util.EventLoopCheckingFuture;
 import com.linecorp.armeria.common.util.SafeCloseable;
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufHolder;
 import io.netty.buffer.Unpooled;
-import io.netty.channel.EventLoop;
 import io.netty.util.ReferenceCountUtil;
 import io.netty.util.concurrent.EventExecutor;
 import io.netty.util.concurrent.ImmediateEventExecutor;
 
 /**
- * Allows subscribing to a {@link StreamMessage} multiple times by duplicating the stream.
- *
- * <p>Only one subscriber can subscribe other stream messages such as {@link DefaultStreamMessage},
- * {@link DeferredStreamMessage}, etc.
- * This factory is wrapping one of those {@link StreamMessage}s and spawns duplicated stream messages
- * which are created using {@link AbstractStreamMessageDuplicator#duplicateStream()} and subscribed
- * by subscribers one by one.</p>
- *
- * <p>The published elements can be shared across {@link Subscriber}s, if you subscribe with the
- * {@link SubscriptionOption#WITH_POOLED_OBJECTS}, so do not manipulate the
- * data unless you copy them.</p>
- *
- * <p>This factory has to be closed by {@link AbstractStreamMessageDuplicator#close()} when
- * you do not need the contents anymore, otherwise memory leak might happen.</p>
+ * A default duplicator.
  *
  * @param <T> the type of elements
- * @param <U> the type of the upstream {@link StreamMessage} and duplicated {@link StreamMessage}s
+ * @see StreamMessageDuplicator
  */
-public abstract class AbstractStreamMessageDuplicator<T, U extends StreamMessage<T>>
-        implements SafeCloseable {
+public class DefaultStreamMessageDuplicator<T> implements StreamMessageDuplicator<T>, SafeCloseable {
```

Comment:
Do we need the `Default*` to be public?

Reply:
Yes, `HttpRequestDuplicator` and `HttpResponseDuplicator` are in different package.
So I was going to move this to the internal package but a lot of other files in the stream package have to come together. I will do it after this is merged. 😄

Reply:
Thanks - sounds like you're signing yourself up for https://github.com/line/armeria/issues/2360 ? 😆

Reply:
Well, I was actually just going to handle the classes under `stream` package.😆
If you want I can take it. 😄 Or maybe we can divide the packages and each takes care of them. Either way is fine for me. 😎